processor/transformer/transformer.go

```diff
@@ -101,6 +101,21 @@ type TransformerEvent struct {
 	Credentials []Credential               `json:"credentials"`
 }
 
+func (e *TransformerEvent) Dehydrate() *TransformerEvent {
```

Comment:
```suggestion
func (e *TransformerEvent) Dehydrate() *HydradedTransformerEvent {
```

Instead of using the same struct, with some fields missing, we can introduce new structs with only the fields we want to include.

This will make it easier to understand, what we plan to send and what not. Plus making the object allocation a bit lighter

Reply:
This would make it a bit difficult downstream as the `transform` function expected `TransformerEvent`. Was planning to do the same in the refactor to reduce the fields further